app/src/main/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonsViewModel.kt

```diff
@@ -1,25 +1,86 @@
 package org.cru.godtools.ui.dashboard.lessons
 
+import android.content.Context
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
 import javax.inject.Inject
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
 import org.cru.godtools.analytics.model.OpenAnalyticsActionEvent
 import org.cru.godtools.analytics.model.OpenAnalyticsActionEvent.Companion.ACTION_OPEN_LESSON
 import org.cru.godtools.analytics.model.OpenAnalyticsActionEvent.Companion.SOURCE_LESSONS
+import org.cru.godtools.base.Settings
+import org.cru.godtools.db.repository.LanguagesRepository
 import org.cru.godtools.db.repository.ToolsRepository
+import org.cru.godtools.db.repository.TranslationsRepository
+import org.cru.godtools.model.Language
+import org.cru.godtools.model.Language.Companion.filterByDisplayAndNativeName
+import org.cru.godtools.ui.dashboard.tools.ToolsScreen.Filters.Filter
 import org.greenrobot.eventbus.EventBus
 
 @HiltViewModel
-class LessonsViewModel @Inject constructor(private val eventBus: EventBus, toolsRepository: ToolsRepository) :
-    ViewModel() {
-    val lessons = toolsRepository.getLessonsFlow()
+class LessonsViewModel @Inject constructor(
+    @ApplicationContext
+    private val context: Context,
+    private val eventBus: EventBus,
+    toolsRepository: ToolsRepository,
+    languagesRepository: LanguagesRepository,
+    private val translationsRepository: TranslationsRepository,
+    val settings: Settings,
+) : ViewModel() {
+
+    val selectedLanguage = MutableStateFlow(Language(settings.appLanguage))
+    fun updateSelectedLanguage(newLanguage: Language) {
+        selectedLanguage.value = newLanguage
+    }
+
+    val query = MutableStateFlow("")
+    fun updateQuery(newQuery: String) {
+        query.value = newQuery
+    }
+
+    private val allLessons = toolsRepository.getLessonsFlow()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val lessonsFlow = selectedLanguage.flatMapLatest { languages ->
```

Comment:
the `languages` variable is an inaccurate name and can lead to confusion. when I first read it I thought it was a list of languages and not just your currently selected language. Rename it to something singular instead of plural

Reply:
you are also only using `lessonsFlow` in one spot within the ViewModel, please simplify by moving the `selectedLanguage.flatMapLatest {}` to where you are using `lessonsFlow`